osquery/utils/aws_util.cpp

```diff
@@ -353,8 +354,8 @@ void getInstanceIDAndRegion(std::string& instance_id, std::string& region) {
   static std::atomic<bool> checked(false);
   static std::string cached_id;
   static std::string cached_region;
-  if (checked) {
-    // Return if already checked
+  if (checked || !isEc2Instance()) {
+    // Return if already checked of it this is not EC2 instance
```

Comment:
The verbage here is a little strange, maybe `// Return if already checked or this is not an EC2 instance`?

Reply:
My bad. Fixing